internal/disk/disk.go

```diff
@@ -184,19 +168,3 @@ func NewVolIDFromRand(r *rand.Rand) string {
 	}
 	return hex.EncodeToString(volid)
 }
-
-func CheckMountpoints(mountpoints []blueprint.FilesystemCustomization, mountpointAllowList *pathpolicy.PathPolicies) error {
```

Comment:
I think this fits better in the new `pathpolicy` package instead of `blueprint` and should have probably moved with the rest of the policy parts.  This shouldn't create any import issues since the `pathpolicy` is imported in `blueprint` only for these checks.

Since the policies are created and managed in the `pathpolicy` package, it seems more natural that the code to use them for the main purpose (policy checking) would be in that same package instead of where the values being checked are defined (`blueprint`).

Reply:
While I agree that policies are currently created and managed in the `pathpolicy` package, It could be applied on any object with paths. So in theory it could be applied also on a slice of `disk.Filesystem` or any other.

In the same sense as we do not refer to `osbuild` package structures in the internal API-agnostic code, I do believe that it is also good to not refer to the `blueprint` package structures outside of it unless necessary to convert them. I get that `blueprint` is kind of API-agnostic concept in composer so comparing it to `osbuild` package is maybe a stretch.

Lastly, my thinking was also to keep all helper functions for `blueprint` structures in that package, so those that transform them to the "internal API" structures, but also those that are doing come validation  / (policy) checks on them.

But I do not feel very strongly about keeping this in the `blueprint` package.

Reply:
I see your point.  Let's keep it the way it is now then, in the `blueprint` package.
I'm coming around to your way of thinking of having the "internal" structures never import or refer to neither `blueprint` nor `osbuild`.